### PR TITLE
Add luminosify color util

### PR DESCRIPTION
**Pull request by benoneal**

### CODE REVIEW

File: packages/rug/src/utils.js [hunk 1]
```diff
@@ -26,10 +34,35 @@ export const calcModularScale = curry((scale, exponent) =>
...
 )
 
+/**
+ * Color Utilities
+ */
+
 export const addUnit = curry((unit, value) => `${value}${unit}`)
 
-export const opacify = (o, rgb) => rgb.replace('b(', 'ba(').replace(')', `,${o})`)
+export const opacify = (rgb, o = 1) => rgb.replace('b(', 'ba(').replace(')', `,${o})`)
```
Comment (lukebrooker) [inline packages/rug/src/utils.js:43]: The signature was initially opacity first and curried to allow the creation of opacity functions.

e.g.
```js
const opacity = {
  light: opacify(0.4)
  lighter: opacify(0.4)
}

style = {
  color: opacity.light(color.charcoal)
}
```

I did like this pattern. Fine if we don't want to go there, but just so you are aware if making this change.

File: packages/rug/src/utils.js [hunk 2]
```diff
@@ -26,10 +34,35 @@ export const calcModularScale = curry((scale, exponent) =>
...
+export const luminosify = (color, amount = 0) => transformLab(modL(amount))(color)
+
+export const gradient = curry((type, degrees, color, lighten = false) => (
+  `${type}-gradient(${degrees}, ${lighten ? color : luminosify(color, -5)}, ${lighten ? luminosify(color, 5) : color})`
+))
 
-export const getContrastColor = ({dark, light}) => ([r, g, b]) => (
-  (((r*299)+(g*587)+(b*114))/1000) <= 128 ? light : dark
+export const shadow = (color) => (
+  `0 4px 6px ${opacify(color, 0.11)}, 0 1px 3px ${opacify(color, 0.08)}`
```
Comment (lukebrooker) [inline packages/rug/src/utils.js:67]: This is not a utility. It's too specific.

I've actually got a shadow utility in my current branch I was going to push soon. We then need a shadows array for different levels.

File: packages/rug/src/values.js [hunk 3]
```diff
@@ -182,6 +184,18 @@ const contrast = (colorName) => (
   getContrastColor({dark: color.charcoal, light: color.white})(brandColors[colorName])
 )
 
+export const gradient = reduce({
+  up: '0deg',
+  right: '90deg',
+  down: '180deg',
+  left: '270deg'
```
Comment (lukebrooker) [inline packages/rug/src/values.js:191]: One of the main gradients in the designs is actually not represented in this. Which is top but slightly left,  ~350deg.

File: packages/rug/src/values.js [hunk 4]
```diff
@@ -197,7 +211,8 @@ export const opacity = {
 
 export const shadows = [
   'none',
-  `0 ${size(1)} ${size(2)} ${size(3)} ${opacify(0.2, color.grey)}`
+  `0 ${size(1)} ${size(2)} ${size(3)} ${opacify(color.border.dark, 0.2)}`,
```
Comment (lukebrooker) [inline packages/rug/src/values.js:214]: Also working on making this a progressive shadow array for different levels.

**Review by lukebrooker — COMMENTED** [review]
Would be good if we had multiple people check rug based PRs before merging.

File: packages/rug/src/utils.js (hunk 1 above)
Comment (benoneal) [reply packages/rug/src/utils.js:43]: At this point I'd strongly prefer to not change the opacity of a color directly at all, ever. `opacify` is really just a helper util for things like shadows.

Comment (lukebrooker) [reply packages/rug/src/utils.js:43]: Shadows are a big part, but it's also one of the only performant ways to transition things, and in this context could also be used a lot.

**Pull request merged**